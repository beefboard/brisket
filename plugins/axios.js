/**
 * When implimented in plugins, this will automatically
 * add our access token to all requests we make via
 * axios
 */
export default function({ $axios, redirect, store }) {
  $axios.onRequest(config => {
    config.headers['x-access-token'] = store.state.token
  })

  $axios.onError(error => {})
}
