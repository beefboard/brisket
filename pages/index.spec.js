import { shallowMount } from '@vue/test-utils'
import index from './index'
import axios from '../plugins/axios'
import flushPromises from 'flush-promises'

jest.mock('axios')

describe('index', () => {
  test('mounts properly', () => {
    const wrapper = shallowMount(index)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = shallowMount(index)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have username input', () => {
    const wrapper = shallowMount(index)
    expect(
      wrapper.html().includes('<input type="text" placeholder="Username">')
    ).toBeTruthy()
  })

  it('should accept login', async () => {
    const wrapper = shallowMount(index)
    wrapper.find('button').trigger('click')
    await flushPromises()
  })
})
