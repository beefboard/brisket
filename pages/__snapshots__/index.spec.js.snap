// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders properly 1`] = `
<section class="container">
  <div>
    <logo-stub></logo-stub>
    <form method="post" class="login"><input type="text" placeholder="Username"> <input type="password" placeholder="Password"> <button>Test</button></form>
  </div>
</section>
`;
